src/pad_calc.cpp

```diff
@@ -86,5 +87,28 @@ std::vector<std::size_t> calc_dyn_auto_pad(std::vector<std::size_t> tensor_lens,
     return padding;
 }
 
+shape compute_padded_shape(const shape& input,
+                           const shape& weights,
+                           const std::vector<std::size_t>& padding,
+                           const std::vector<std::size_t>& stride,
+                           const std::vector<std::size_t>& dilation)
+{
+    const size_t num_spatial_dims = input.lens().size() - 2;
+
```

Comment:
can you add a check here incase somehow num_spatial_dims roll over? size_t I don't think goes negative.

```
if (incase num_spatial_dims > input.lens().size()) 
                  MIGRAPHX_THROW(...)
'''

Reply:
The way this function is supposed to be used, the rank of `input` should always be `>= 4`. Should this be an assert or a throw?